backend/routers/spiritual.py

```
@@ -789,4 +792,100 @@ def get_default_lord(house_index: int) -> str:
+        
+        enhanced_service = EnhancedBirthChartCacheService()
+        complete_profile = await enhanced_service.generate_and_cache_complete_profile(user_email, birth_details)
+        
+        return {
+            "success": True,
+            "complete_profile": complete_profile
+        }
+        
+    except Exception as e:
+        logger.error(f"Error generating birth chart for user: {e}")
+        raise HTTPException(status_code=500, detail="Failed to generate birth chart") 
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)